### ISSUE TITLE
rpc_client.pending error

### ISSUE THREAD

**Issue by pawapps** [appditto/pippin_nano_wallet#1]
https://github.com/appditto/pippin_nano_wallet/blob/0bfb9fb03901661094dd4cfa957925443a721ded/pippin/network/rpc_client.py#L118

Data passed in is to above line of code is:  `resp = {'blocks': ['1234...', 'ABCD...']}`

Errors when iterating over .items()

Remove `for k, v` loop and simply `return resp['blocks']`

**Comment by pawapps** [comment]
Just upgraded from 1.0.8 -> 1.1.11. Still have this bug and had to redo the changes in the rpc_client.py.  Here's some additional log data if helpful:

`pip show pippin-wallet`

> Name: pippin-wallet
> Version: 1.1.11
> Summary: A production-ready, high-performance developer wallet for Nano and BANANO.
> Home-page: https://github.com/appditto/pippin_nano_wallet
> Author: Appditto LLC
> Author-email: hello@appditto.com
> License: MIT License
> Location: /home/jason/.local/lib/python3.7/site-packages
> Requires: aioredlock, pycryptodome, bitstring, aiounittest, nanolib, aiodns, aiohttp, setuptools, pypika, aiosqlite, python-dotenv, python-rapidjson, uvloop, websockets, nanopy, aiomysql, aioredis, asyncpg, tortoise-orm, pyyaml
> Required-by:

From the log

> May 17 18:50:08 SomeNanoNode pippin-server[26185]: ERROR:aiohttp.server:Error handling request
> May 17 18:50:08 SomeNanoNode pippin-server[26185]: Traceback (most recent call last):
> May 17 18:50:08 SomeNanoNode pippin-server[26185]:   File "/home/jason/.local/lib/python3.7/site-packages/aiohttp/web_protocol.py", line 418, in start
> May 17 18:50:08 SomeNanoNode pippin-server[26185]:     resp = await task
> May 17 18:50:08 SomeNanoNode pippin-server[26185]:   File "/home/jason/.local/lib/python3.7/site-packages/aiohttp/web_app.py", line 458, in _handle
> May 17 18:50:08 SomeNanoNode pippin-server[26185]:     resp = await handler(request)
> May 17 18:50:08 SomeNanoNode pippin-server[26185]:   File "/home/jason/.local/lib/python3.7/site-packages/aiohttp/web_middlewares.py", line 119, in impl
> May 17 18:50:08 SomeNanoNode pippin-server[26185]:     return await handler(request)
> May 17 18:50:08 SomeNanoNode pippin-server[26185]:   File "/home/jason/.local/lib/python3.7/site-packages/aiohttp/web_middlewares.py", line 109, in impl
> May 17 18:50:08 SomeNanoNode pippin-server[26185]:     return await handler(request)
> May 17 18:50:08 SomeNanoNode pippin-server[26185]:   File "/home/jason/.local/lib/python3.7/site-packages/pippin/server/pippin_server.py", line 79, in gateway
> May 17 18:50:08 SomeNanoNode pippin-server[26185]:     return await self.send(request, request_json)
> May 17 18:50:08 SomeNanoNode pippin-server[26185]:   File "/home/jason/.local/lib/python3.7/site-packages/pippin/server/pippin_server.py", line 344, in send
> May 17 18:50:08 SomeNanoNode pippin-server[26185]:     resp = await wallet.send(int(request_json['amount']), request_json['destination'], id=id, work=work)
> May 17 18:50:08 SomeNanoNode pippin-server[26185]:   File "/home/jason/.local/lib/python3.7/site-packages/pippin/util/wallet.py", line 199, in send
> May 17 18:50:08 SomeNanoNode pippin-server[26185]:     state_block = await self._send_block_create(amount, destination, id=id, work=work)
> May 17 18:50:08 SomeNanoNode pippin-server[26185]:   File "/home/jason/.local/lib/python3.7/site-packages/pippin/util/wallet.py", line 143, in _send_block_create
> May 17 18:50:08 SomeNanoNode pippin-server[26185]:     await self._receive_all()
> May 17 18:50:08 SomeNanoNode pippin-server[26185]:   File "/home/jason/.local/lib/python3.7/site-packages/pippin/util/wallet.py", line 115, in _receive_all
> May 17 18:50:08 SomeNanoNode pippin-server[26185]:     p = await RPCClient.instance().pending(self.account.address, threshold=config.Config.instance().receive_minimum)
> May 17 18:50:08 SomeNanoNode pippin-server[26185]:   File "/home/jason/.local/lib/python3.7/site-packages/pippin/network/rpc_client.py", line 118, in pending
> May 17 18:50:08 SomeNanoNode pippin-server[26185]:     for k,v in resp['blocks'].items():
> May 17 18:50:08 SomeNanoNode pippin-server[26185]: AttributeError: 'list' object has no attribute 'items'

My changes

>     async def pending(self, account: str, threshold: int) -> List[str]:
>         """RPC Pending"""
>         pending_action = {
>             'action': 'pending',
>             'account': account,
>             'threshold': str(threshold)
>         }
>         resp = await self.make_request(pending_action)
>         ret = []
>         if 'blocks' in resp:
>             if resp['blocks'] == '':
>                 return []
>             return resp['blocks']
>             #for k,v in resp['blocks'].items():
>             #    ret.append(k)
>             #return ret
>         return None

**Comment by bbedward** [comment]
I think the difference in format is related to threshold, if threshold is provided it returns a map otherwise it returns a list. Probably need a little bit if logic to change return when threshold isn't provided

**Comment by bbedward** [comment]
@pawapps could you share your config.yaml and RPC request that triggers the error.

**Comment by pawapps** [comment]
@bbedward -- agreed, did a little testing and saw the same with setting various "receive_minimum" settings.

```
# When no pending meets threshold
{'blocks': ''}

# Pending exists and Setting receive_minimum: 1000000000
{'blocks': {'22B08926...': '10000000000000000000'}}

# Pending exists and Setting receive_minimum: 0
{'blocks': ['43CF...']}

# Pending exists and Commenting out receive_minimum:
{'blocks': {'94366...': '100000000000000000000000000'}}
```

Here's my config.yaml when I had the crashes:
```
# Settings for the pippin server
server:
  host: 127.0.0.1
  port: 11338
  log_file: /tmp/pippin_wallet.log
  node_rpc_url: http://127.0.0.1:7076
  node_ws_url: ws://[::1]:7078
wallet:
  receive_minimum: 0
  auto_receive_on_send: true
```

**Comment by bbedward** [comment]
Addressed in 717d08ac619200ebc3f05e9cea262fe19323678e and published 1.1.12 to pypi, let me know if that works for you

**Comment by pawapps** [comment]
works great, thanks!
